src/peep/peep.c

```diff
@@ -7212,7 +7212,7 @@ static int sub_69A997(sint16 x, sint16 y, uint8 z, uint8 counter, uint16 score,
 			footpath_element_get_slope_direction(path) == test_edge) {
 			height += 2;
 		}
-		sub_69A997(x, y, height, counter, score, test_edge);
+		score = sub_69A997(x, y, height, counter, score, test_edge);
```

Comment:
@duncanspumpkin While the other line was a significant part of why this patch was needed, it was this line that caused the majority of the issues :P (just in case you were wondering).